CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+    2. [Commits](#commits)
+    3. [Pull requests](#pull-requests)
+        1. [Reviewing pull requests](#reviewing-pull-requests)
+        2. [Merging pull requests](#merging-pull-requests)
+
+## Commits and pull requests
+### Fixing a bug
+
+1. Fork the repo and commit changes to a new branch in your fork.
+2. Make pull requests against the master branch in the rpc-gating repo.
+3. Reference the JIRA issue in both the commit message and the pull request message.
+
+### Commits
+
+Please avoid the following in a commit:
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
+
+Expected git commit message structure:
+
+* The first line should be limited to 50 characters, should not end with a full stop and the first word should be capitalised.
+* Use the imperative mood in the subject line - e.g. ```Fix a typo in CONTRIBUTING.md```, ```Remove if/else block in myfile.sh```, ```Eat your dinner```.
+* Insert a single blank line after the first line.
+* Subsequent lines should be wrapped at 72 characters.
+* Provide a detailed description of the change in the following lines, using the guidelines in the section below.
+* The last line of the commit should be a reference to the issue being fixed using the JIRA issue Key e.g. ```RE-1500```
+
+In your commit message please consider the following points:
+
+* In the commit subject line, provide a brief description of the change.
+* The commit message must be descriptive and contain all the information required to fully understand and review the patch.
+* Do not assume the reviewer understands what the original problem was.
+* Do not assume the reviewer has access to external web services or site.
+* Do not assume the code is self-evident or self-documenting.
+* Describe why the change is being made.
+* If the commit message suggests an improved code structure, you may be able to split this into two or more commits.
+* Ensure sufficient information is provided to review your commit.
+
+
+### Pull requests
+
+* A pull request (PR) should ideally contain a single commit
+* The PR title and message should contain information relevant to why the commit(s) are happening. This can be based off the commit message.
+* The PR description should reference the original issue.
+* Where absolutely necessary, related commits can be grouped into a single PR, but this should be the exception, not the rule.
+
+#### Reviewing pull requests
+
+When reviewing a PR, please ensure the included commit(s):
+
+* Actually works to fix the issue.
+* Passes the gate checks that are configured to run.
+* Contains a single commit, which has a commit message describing the changes involved in the patch. On a rare occasion, multiple related commits in the same PR could be considered.
+* Does not overreach. Each commit should be self contained to only address the issue at hand (see above).
```

Comment:
Are we really wanting to kill any technical debt management which often comes as a side-effect to working on specific problems? For example, sometimes while working on a problem you notice that the problem is easier to solve by doing some simplification of functions which touch other things that don't directly relate to the problem. As such, technical debt is managed as part of the issue at hand, even though some of that is unrelated. Applying this rule will reduce that effect, making it more likely that we grow tech debt instead of continually reduce it where possible.

Reply:
I think we can all use our heads when reviewing and making sure that any refactoring is not excessive and unrelated.